core/trino-main/src/main/java/io/trino/server/ServerMainModule.java

```diff
@@ -443,6 +443,8 @@ protected void setup(Binder binder)
         discoveryBinder(binder).bindHttpAnnouncement("trino")
                 .addProperty("node_version", nodeVersion)
                 .addProperty("coordinator", String.valueOf(serverConfig.isCoordinator()));
+        // TODO replace with dynamic properties support in io.airlift.discovery.client.ServiceAnnouncement.ServiceAnnouncementBuilder
+        binder.bind(AnnounceNodeInstanceId.class).asEagerSingleton();
```

Comment:
This is obviously a hack that works around lack of support for guice-level dynamic properties in HTTP announcement builder.
Having something like https://github.com/airlift/airlift/pull/1157 should allow to avoid this.